### PR TITLE
Refact virtualize.sh

### PR DESCRIPTION
**Pull request by cedlecomte**
- Refactoring the virtualize.sh with many parameters
- Adding the socks option to create a socks server on the hypervisor to join your platform
- Adding the tempest option to launch sanity job after a deployment

### CODE REVIEW

**Comment by ErwanAliasr1** [comment]
How do you motivate removing @fredericlepied  from the authors ?

**Comment by ErwanAliasr1** [comment]
That's not really my project, but I'm not that fan of a commit that change all that amount of stuff in a single commit.

**Comment by cedlecomte** [comment]
The goal was to add myself not to delete fred.

**Comment by goneri** [comment]
@ErwanAliasr1  We prefer to avoid those author headers. See hacking err H105 for a full justification. Basically, these lines are not legally mandatory, create merge conflict without benefit, can create frustration between contributors (e.g: Why I'm not in the Author header?) and are supersede by `git log` and `git blame`.
@ErwanAliasr1 yep, the commit may looks a bit large but it's mostly re-factoring, including indentation changes.
@Prophidys can you increase a bit the size of the commit message to give an overview of the new changes?

File: virtualize.sh [hunk 1]
```diff
@@ -16,80 +16,49 @@
...
-SpinalStack environment to deploy. It will then upgrade the newly deployed
-SpinalStack to the following environment directory.
-
-For example: ./virtualize.sh I.1.2.1 I.1.3.0 I.1.3.1
-will deploy environment I.1.2.1 and upgrade to I.1.3.0 and then I.1.3.1."
-      exit 1
+    -H|--hypervisor=name          Set the hypervisor hostname, default (${virthost})
+    -d|--debug                    Set the debug mode for this script, default: disabled
+    -w|--wordkir=dir1,dir2,...    Workdir List, default: None
+    -v|--virt=virt_platform.yml   Set the path to the infra's yaml
```
Comment (goneri) [inline virtualize.sh]: i don't see the benefit of this new parameter. So far, the virt_playtform.yml was in the workdir. I don't understand why you need to change that.

File: virtualize.sh [hunk 2]
```diff
@@ -110,90 +79,104 @@ upload_logs() {
...
+            ssh $SSHOPTS root@${virthost} virsh destroy ${PREFIX}_${installserver_name}
+            for snapshot in $(ssh $SSHOPTS root@${virthost} virsh snapshot-list --name goneri_${installserver_name}); do
+                ssh $SSHOPTS root@${virthost} virsh snapshot-delete ${PREFIX}_${installserver_name} ${snapshot}
             done
-            ssh $SSHOPTS root@$virthost virsh undefine --remove-all-storage ${PREFIX}_${installserver_name}
+            ssh $SSHOPTS root@${virthost} virsh undefine --remove-all-storage ${PREFIX}_${installserver_name}
             jenkins_job_name="upgrade"
         fi
     else
-#        virtualizor_extra_args="--replace"
```
Comment (goneri) [inline virtualize.sh:128]: This line should be dropped then.

File: virtualize.sh [hunk 3]
```diff
@@ -16,80 +16,49 @@
...
-will deploy environment I.1.2.1 and upgrade to I.1.3.0 and then I.1.3.1."
-      exit 1
+    -H|--hypervisor=name          Set the hypervisor hostname, default (${virthost})
+    -d|--debug                    Set the debug mode for this script, default: disabled
+    -w|--wordkir=dir1,dir2,...    Workdir List, default: None
+    -v|--virt=virt_platform.yml   Set the path to the infra's yaml
+    -e|--extra='--replace'        Add extra parameters to virtulizor.py
+    -s|--socks                    Create a socks server to test your platform
+    -p|--prefix                   Change the platform's prefix, default: unix user
+    -t|--tempest                  Launch the sanity job at the end of a deployement
```
Comment (goneri) [inline virtualize.sh:50]: this is an enhancement and so should be introduced by another commit, actually, that also the case of the `--socks` parameter.

File: virtualize.sh (hunk 1 above)
Comment (cedlecomte) [inline virtualize.sh]: I will add a default value but it will be nice to have this as a parameter to test different files.

File: virtualize.sh [hunk 4]
```diff
@@ -16,80 +16,51 @@
...
-SpinalStack to the following environment directory.
-
-For example: ./virtualize.sh I.1.2.1 I.1.3.0 I.1.3.1
-will deploy environment I.1.2.1 and upgrade to I.1.3.0 and then I.1.3.1."
-      exit 1
+    -h|--help                     Show this help
+    -H|--hypervisor=name          Set the hypervisor hostname, default (${virthost})
+    -d|--debug                    Set the debug mode for this script, default: disabled
+    -w|--wordkir=dir1,dir2,...    Workdir List, default: None
+    -v|--virt=virt_platform.yml   Set the path to the infra's yaml, default: virt_platform.yaml
```
Comment (goneri) [inline virtualize.sh:46]: As said on HG, I would prefer the virt_platform.yml to be in the current $workdir.
